apps/roam/src/utils/getLeftSidebarSettings.ts

```diff
@@ -0,0 +1,175 @@
+  BooleanSetting,
+  getUidAndBooleanSetting,
+  getUidAndStringSetting,
+  IntSetting,
+  getUidAndIntSetting,
+  StringSetting,
+} from "./getExportSettings";
+import { getSubTree } from "roamjs-components/util";
+import getCurrentUserDisplayName from "roamjs-components/queries/getCurrentUserDisplayName";
+
+type LeftSidebarPersonalSectionSettings = {
+  uid: string;
+  truncateResult: IntSetting;
+  folded: BooleanSetting;
+  alias: StringSetting | null;
+};
+
+export type LeftSidebarPersonalSectionConfig = {
+  uid: string;
+  text: string;
+  sectionWithoutSettingsAndChildren: boolean;
+  settings?: LeftSidebarPersonalSectionSettings;
+  children?: RoamBasicNode[];
+  childrenUid?: string;
+};
+
+type LeftSidebarGlobalSectionConfig = {
+  uid: string;
+  collapsable: BooleanSetting;
+  folded: BooleanSetting;
+  children: RoamBasicNode[];
+  childrenUid: string;
+};
+
+export type LeftSidebarConfig = {
+  global: LeftSidebarGlobalSectionConfig;
+  personal: {
+    uid: string;
+    sections: LeftSidebarPersonalSectionConfig[];
+  };
+};
+
+const getLeftSidebarGlobalSectionConfig = (
+  leftSidebarChildren: RoamBasicNode[],
+): LeftSidebarGlobalSectionConfig => {
+  const globalSectionNode = getSubTree({
+    tree: leftSidebarChildren,
+    key: "Global-Section",
+  });
+  const globalChildren = globalSectionNode?.children || [];
+
+  const childrenNode = getSubTree({
+    tree: globalChildren,
+    key: "Children",
+  });
+
+  return {
+    uid: globalSectionNode?.uid || "",
+    collapsable: getUidAndBooleanSetting({
+      tree: globalChildren,
+      text: "Collapsable",
+    }),
+    folded: getUidAndBooleanSetting({ tree: globalChildren, text: "Folded" }),
+    children: childrenNode?.children || [],
+    childrenUid: childrenNode?.uid || "",
+  };
+};
+
+const getPersonalSectionSettings = (
+  settingsNode: RoamBasicNode,
+): LeftSidebarPersonalSectionSettings => {
+  const settingsTree = settingsNode?.children || [];
+
+  const truncateResultSetting = getUidAndIntSetting({
+    tree: settingsTree,
+    text: "Truncate-result?",
+  });
+
+  if (truncateResultSetting.value === 0) {
+    truncateResultSetting.value = 75;
+  }
+  const foldedSetting = getUidAndBooleanSetting({
+    tree: settingsTree,
+    text: "Folded",
+  });
+
+  const aliasString = getUidAndStringSetting({
+    tree: settingsTree,
+    text: "Alias",
+  });
+  const alias =
+    aliasString.value === "Alias" ||
+    (aliasString.value === "" && aliasString.uid)
+      ? null
+      : aliasString;
+
+  return {
+    uid: settingsNode.uid,
+    truncateResult: truncateResultSetting,
+    folded: foldedSetting,
+    alias,
+  };
+};
+
+const getLeftSidebarPersonalSectionConfig = (
+  leftSidebarChildren: RoamBasicNode[],
+): { uid: string; sections: LeftSidebarPersonalSectionConfig[] } => {
+  const userName = getCurrentUserDisplayName();
+
+  const personalLeftSidebarNode = getSubTree({
+    tree: leftSidebarChildren,
+    key: userName + "/Personal-Section",
```

Comment:
We should have this on the page as `[[userName]]` so that when the username changes, so does this setting.

Reply:
good catch

Reply:
`getSubTree` does not recognize if we use double brackets I tried 
- `"[[" + userName + "]]/Personal-Section"`
- `"\\[\\[" + userName + "\\]\\]/Personal-Section",`
- `"\\\\[\\\\[" + userName + "\\\\]\\\\]/Personal-Section"`

Reply:
you mean instead of using username for storage and comparision use the uid?